blockchain/pool.go

```diff
@@ -452,9 +453,9 @@ func (bpr *bpRequester) getPeerID() string {
 
 func (bpr *bpRequester) reset() {
 	bpr.mtx.Lock()
+	defer bpr.mtx.Unlock()
```

Comment:
Really minor comment that can be ignored: so at times, while defer can be more
convenient to read, it can be slower but idk if this matters